### ISSUE TITLE
Regular pre-training performance spikes

### ISSUE THREAD

**Issue by siemdejong** [siemdejong/sclicom#26]
**Describe the bug**
The performance of the pre-trainer simclr spikes at regular 1900 steps, which is a pattern that cannot be ignored. Since the best model is saved (based on the lowest training loss), the best model becomes useless as the loss makes no sense.

![image](https://user-images.githubusercontent.com/28396796/226099372-87d0dc15-fe74-4025-8c84-8939fc63d9d9.png)

**To Reproduce**
1. Train a SimCLR with PMCHHGImageDataModule for at least 6000 steps to see the pattern.
2. Batch size 1024

**Expected behavior**
Smooth loss.

**Desktop (please complete the following information):**
 - OS: rocky linux
 - DPAT version 3.2.1.

**Additional context**
[Add any other context about the problem here.](https://stackoverflow.com/questions/60466189/cnn-training-loss-regular-spikes-at-the-end-of-the-epoch?noredirect=1&lq=1) and linked issues.

Looks like it can be the loss reset at every epoch. Either don't do that and fix it to do something else, or use `drop_last` in the dataloader, such that the model has not varying batch size.

It's a stretch, but it could also be an issue of torch2.0 compile, where the loss function is traced with a large batch size and reused while the batch size is small for the last minibatch of an epoch.

Either scenarios do not explain why it does not happen _every_ epoch, though. It only happend 6 times for 30 epochs.

**Comment by siemdejong** [comment]
Drop last fixed this. Apparently the loss wasn't calculated right when the last batch was of different size.
